### PR TITLE
Migrating to Androidx 

### PR DESCRIPTION
**Pull request by temirnurdin**
* Update support library to androidx

### CODE REVIEW

**Comment by temirnurdin** [comment]
@thyrlian Ping if anything additional required.

File: demo/src/main/res/layout/activity_demo.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
         android:id="@+id/toolbar"
         layout="@layout/toolbar"/>
 
-    <android.support.v4.widget.DrawerLayout
+    <androidx.drawerlayout.widget.DrawerLayout
         xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thyrlian) [inline demo/src/main/res/layout/activity_demo.xml:15]: This namespace declaration is redundant, can be removed.  But this was my bad.

File: gradle.properties [hunk 2]
```diff
@@ -33,4 +33,5 @@ pomLicenceName=The MIT License (MIT)
 pomLicenceUrl=http://opensource.org/licenses/MIT
 pomLicenceDist=repo
 pomDeveloperId=thyrlian
-pomDeveloperName=Jing Li
\ No newline at end of file
+pomDeveloperName=Jing Li
+android.useAndroidX=true
```
Comment (thyrlian) [inline gradle.properties:37]: It's better to have segregation with the above POM attributes.  But this is a minor thing.

File: library/build.gradle [hunk 3]
```diff
@@ -46,15 +46,15 @@ android {
 
 ext {
     guavaVersion = "29.0-android"
-    junitVersion = "4.13"
-    mockitoVersion = "3.5.7"
+    junitVersion = "4.13.1"
+    mockitoVersion = "3.5.11"
```
Comment (thyrlian) [inline library/build.gradle:50]: Actually, the upgrade of other libraries should come with a different PR, as this PR aims to migrate to Androidx.  Let's leave it as this time, luckily all tests were passed on CI and also locally :)

**Review by thyrlian — APPROVED** [review]
All look good!  Thanks for your effort 👍

**Pull request merged**